addons/barcodes_gs1_nomenclature/static/src/js/epc_utils.js

```diff
@@ -0,0 +1,282 @@
+            return getAi00(model);
+        case "01":
+            return getAi01(model);
+        case "414":
+            return getAi414(model);
+        default:
+            throw new Error(
+                `No extrapolation function found for the GS1 Application Identifier (${identifier})`
+            );
+    }
+}
+
+function getAi00(model) {
+    const companyPrefix = model.fields["company_prefix"];
+    const serialInteger = model.fields["serial_integer"];
+
+    const extensionDigit = serialInteger.value.substr(0, 1);
+    const serialRefRemainder = serialInteger.value.substr(1);
+    const ssccToCheck = extensionDigit + companyPrefix.value + serialRefRemainder;
+    const checkdigit = getGs1Checksum(ssccToCheck);
+    return " (00) " + ssccToCheck + checkdigit;
+}
+
+function getAi01(model) {
+    // input length : sum of companyPrefix.digits and itemReference.digits is always 13 digits (preserving leading zeroes)
+    // output: 14 digits
+    const companyPrefix = model.fields["company_prefix"];
+    const itemReference = model.fields["item_reference"];
+
+    const indicator = itemReference.value.substr(0, 1);
+    const itemRefRemainder = itemReference.value.substr(1);
+    const gtinToCheck = indicator + companyPrefix.value + itemRefRemainder;
+    const checkdigit = getGs1Checksum(gtinToCheck);
+    return " (01) " + gtinToCheck + checkdigit;
+}
+
+function getAi414(model) {
+    const companyPrefix = model.fields["company_prefix"];
+    const locationReference = model.fields["location_reference"];
+
+    // company_prefix + location_ref
+    const glnToCheck = companyPrefix.value + locationReference.value;
+    const checkdigit = getGs1Checksum(glnToCheck);
+    return " (414) " + glnToCheck + checkdigit;
+}
+
+function getGs1Checksum(data) {
+    // The GS1 specification defines the check digit calculation as follows (cf. [TDS 2.1] §7.3):
+    // d14 = (10 - ((3(d1 + d3 + d5 + d7 + d9 + d11 + d13) + (d2 + d4 + d6 + d8 + d10 + d12)) mod 10) mod 10
+    // However, it should be noted that here the index starts at 0 and not 1.
+    // It is therefore necessary to multiply even numbers instead of odd numbers.
+
+    // Note : same method is used for multiple data structures up to 17 digits (excluding check digits).
+    // Cf. https://ref.gs1.org/standards/genspecs/ §7.9.1
+    // data = data.padStart(17, '0'); is not optimal so we cheat on i value of incoming for loop instead
+    const offset = 17 - data.length;
+    let sum = 0;
+    for (let i = 0; i < data.length; i++) {
+        const value = parseInt(data.charAt(i));
+        if ((i + offset) % 2 == 0) {
+            sum += value * 3;
+        } else {
+            sum += value;
+        }
+    }
+    return ((10 - (sum % 10)) % 10).toString();
+}
+
+//public
+export function isValidEpcFormat(hexString) {
+    if (hexString.length < 2) return false;
+    const headerString = hexString.substring(0,2);
+    if (!REGEX_IS_HEXA.test(headerString)) return false;
```

Comment:
A little bit of breathing please :relieved: 
```suggestion
    if (hexString.length < 2) {
        return false;
    }
    const headerString = hexString.substring(0,2);
    if (!REGEX_IS_HEXA.test(headerString)) {
        return false;
    }
```